galsim/phase_psf.py

```diff
@@ -1579,6 +1580,10 @@ def _shoot(self, photons, rng):
         u = photons.pupil_u
         v = photons.pupil_v
         t = photons.time
+        if photons.hasAllocatedWavelengths():
+            w = photons.wavelength
```

Comment:
Need a test that uses this feature.  (Which is the feature request that triggered this PR...)  Maybe one that compares this implementation to one using ChromaticAtmosphere?

Reply:
Yeah, I had naïvely thought that adding a `WavelengthSampler` photon op to `drawImage` would trigger this, but I guess those ops only get executed after the base object is drawn.  Looking into how to actually trigger this now...

Reply:
It's a little bit of a design conundrum even: `PhaseScreenPSF` is currently a `GSObject` (and not a `ChromaticObject`).  That's perfectly fine for the original usage, and would continue to be fine if `seeing_exp = 0`.  But the profile _is_ chromatic, of course, if `seeing_exp != 0`.  Duck-typing might work here, but may require changing some of the code around https://github.com/GalSim-developers/GalSim/blob/main/galsim/gsobject.py#L419.  For instance, the chromatic version of this profile is not separable.

Reply:
Should we just be using ChromaticAtmosphere then?

Reply:
>Should we just be using ChromaticAtmosphere then?

Probably not; ChromaticAtmosphere is something quite different.  It just applies some chromatic transformations to a base object (e.g. Moffat, Kolmogorov, etc) consistent with DCR and chromatic seeing, but doesn't do anything with phase screens.

Reply:
Can't the base object of ChromaticAtmosphere be a PhaseScreenPSF?

Reply:
Hmmm... Maybe?  Let me try.

Reply:
Okay, that does appear to be close.  The main difference is that ChromaticAtmosphere applies its shift to the entire base object at once.  So we couldn't specify an exponent per screen.  This is probably good enough for the atmospheric screens, but won't be correct for any OpticalScreens and I don't think will be correct for the second_kick component of the atmosphere (which has a limiting case of an Airy, e.g.).

Reply:
Can we do something a little custom in ChromaticAtmosphere to make it work right when the base is a PhaseScreenPSF?

Reply:
Yeah.  That or tweak https://github.com/LSSTDESC/imSim/blob/main/imsim/atmPSF.py#L83 to build 3 separate things (atm+opt+2kick) instead of embedding them all inside a single PhaseScreenPSF.  Then we can use ChromaticAtmosphere to only add chromatic seeing to the right part.

Reply:
The latter is probably simpler...